code/implementation/docker/storage/src/main/java/io/cattle/platform/docker/storage/DockerImageCredentialLookup.java

```diff
@@ -0,0 +1,56 @@
+package io.cattle.platform.docker.storage;
+
+import io.cattle.platform.core.model.Credential;
+import io.cattle.platform.core.model.StoragePool;
+import io.cattle.platform.docker.client.DockerImage;
+import io.cattle.platform.storage.ImageCredentialLookup;
+import io.cattle.platform.util.type.CollectionUtils;
+
+import java.util.List;
+
+public class DockerImageCredentialLookup implements ImageCredentialLookup {
+
+    private static final String REGISTRY = "registry";
+    private static final String REGISTRY_CREDENTIAL = "registryCredential";
+
+    @Override
+    public Credential getDefaultCredential(String uuid, List<?> storagePools, List<?> credentials) {
+        String serverAddress = DockerImage.parse(uuid).getServer();
```

Comment:
DockerImage.parse() seems like it could return a null in various cases, so we might want to check for that unless you're absolutely positive this can't happen ;-)